cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/azure_onboarder.py

```diff
@@ -0,0 +1,220 @@
+import argparse
+import configparser
+import logging
+import os
+import sys
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Tuple
+
+from az.cli import az
+from python_terraform import IsFlagged, IsNotFlagged, Terraform
+
+log = logging.getLogger(__name__)
+logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.WARNING)
+EX_OK: int = 0  # exit code for successful command
+EX_FAILED: int = 1  # exit  code for failed command
+PROFILES_FILE: str = "profiles.ini"
+
+
+# pylint: disable=too-many-instance-attributes
+@dataclass
+class AzureProfile:
+    name: str
+    subscription_id: str
+    tenant_id: str
+    principal_id: str
+    principal_secret: str
+    location: str
+    resource_group_names: List[str]
+    storage_account_names: List[str]
+
+
+TerraformVars = Dict[str, Any]  # variables passed in terraform plan/apply/destroy call
+TerraformAction = Callable[[Terraform, TerraformVars], bool]  # terraform plan/apply/destroy
+
+# execute an action for every profile
+def execute_action(action: TerraformAction, profiles: List[AzureProfile]) -> bool:
+    t = Terraform()
+    successful_count = 0
+    for profile in profiles:
+        print(f'Profile: "{profile.name}" ({profile.location})')
```

Comment:
I think that all print calls should be changed to log.Debug and log.Info calls for consistency with warning/critical logs.

Reply:
Will separate user-targeted printouts from logs recording how the application execution goes. Will redirect logs to file to keep user screen clean while collecting enough information for troubleshooting.

Reply:
Updated logging and user communication:

- user gets only high level information on process execution
- detailed information is logged to file